phoenix-core/src/it/java/org/apache/phoenix/end2end/AuditLoggingIT.java

```diff
@@ -65,91 +68,117 @@ public void testLoggingSelect() throws Exception {
         String getLogsQuery = "select * from SYSTEM.LOG WHERE TABLE_NAME='TEST2' order by start_time";
         Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
         props.setProperty(QueryServices.LOG_LEVEL, LogLevel.TRACE.name());
-        Connection conn = DriverManager.getConnection(getUrl(), props);
-        conn.setAutoCommit(true);
-        try {
-            conn.createStatement().execute(createqQery);
-            conn.createStatement().execute(upsertQuery);
-            ResultSet rs = conn.createStatement().executeQuery(selectQuery);
+        try (Connection conn = DriverManager.getConnection(getUrl(), props);
+                Statement stmt = conn.createStatement();
+        ){
+            conn.setAutoCommit(true);
+            stmt.execute(createqQery);
+            stmt.execute(upsertQuery);
+            ResultSet rs = stmt.executeQuery(selectQuery);
             assertTrue(rs.next());
             assertFalse(rs.next());
             rs.close();
 
-            ResultSet rs2 = conn.createStatement().executeQuery(getLogsQuery);
-            assertTrue(rs2.next());
-            assertEquals(rs2.getString(7), selectQuery);
-            assertFalse(rs2.next());
-        } finally {
-            conn.close();
+            int retryCount=10;
+            while (true) {
+                try {
+                    ResultSet rs2 = stmt.executeQuery(getLogsQuery);
+                    assertTrue(rs2.next());
+                    assertEquals(rs2.getString(7), selectQuery);
+                    assertFalse(rs2.next());
+                    break;
+                } catch (AssertionFailedError e) {
```

Comment:
Shouldn't we catch AssertionError instead of AssertionFailedError?

Reply:
Thanks, you are right.
Fixed.